sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs

```diff
@@ -38,12 +37,35 @@ public static HttpPipeline Build(ClientOptions options, params HttpPipelinePolic
             ClientOptions options,
             HttpPipelinePolicy[] perCallPolicies,
             HttpPipelinePolicy[] perRetryPolicies,
-            ResponseClassifier? responseClassifier)
+            ResponseClassifier? responseClassifier = default)
         {
             var result = BuildInternal(options, perCallPolicies, perRetryPolicies, null, responseClassifier);
             return new HttpPipeline(result.Transport, result.PerCallIndex, result.PerRetryIndex, result.Policies, result.Classifier);
         }
 
+        /// <summary>
+        /// </summary>
+        /// <param name="options"></param>
+        /// <param name="perCallPolicies"></param>
+        /// <param name="perRetryPolicies"></param>
+        /// <param name="messageClassifier">Classifier to be applied during classification for every operation on the client.</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException">Thrown when the <paramref name="messageClassifier"/> is null.</exception>
+        public static HttpPipeline Build(
+            ClientOptions options,
+            HttpPipelinePolicy[] perCallPolicies,
+            HttpPipelinePolicy[] perRetryPolicies,
+            MessageClassifier messageClassifier)
+        {
+            Argument.AssertNotNull(messageClassifier, nameof(messageClassifier));
+
+            CompositeClassifier classifier = new CompositeClassifier(ResponseClassifier.Shared);
```

Comment:
Would there ever be a scenario where someone needs a custom baseline classifier in addition to the messageClassifer?  Seems unlikely, but if previously someone was passing one to the other overload, they lose that capability here.

Reply:
Yes, we allow that by having ResponseClassifier be settable on HttpMessage.  You can see that DPG clients do that always: https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/webpubsub/Azure.Messaging.WebPubSub/src/Generated/WebPubSubServiceClient.cs#L1529  
That is why we needed to introduce the tray semantics on MessageClassifier, so we could fall through to the base classifier, but the base classifier is pluggable too (in DPG we have a different base classifier per operation).

Reply:
The other fun part here is that base classifiers are typically static shared instances, so we can't modify them directly -- that's why we needed `CompositeClassifier` instead of making modifications to `ResponseClassifier` directly.

Reply:
I looked briefly at adding this, but didn't want to add the docstring for the parameter saying, actually, a lot of the time we're going to override this, so don't pass anything in for this parameter.